tests/deriver/test_representation_crud.py

```
@@ -106,3 +108,25 @@ def test_prompt_representation_conversion():
+    nested = PromptRepresentation.model_validate({"result": {"observations": ["I work remotely"]}})
+    assert [item.content for item in nested.explicit] == ["I work remotely"]
+
+
+def test_prompt_representation_truncates_overlong_content():
+    prompt_rep = PromptRepresentation.model_validate({"explicit": [{"content": "x" * 5000}]})
+    assert len(prompt_rep.explicit) == 1
+    assert len(prompt_rep.explicit[0].content) == 2000
+
+
+def test_prompt_representation_rejects_unknown_top_level_dict_shape():
+    with pytest.raises(Exception):
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)